spec/index.bs

```diff
@@ -2612,6 +2613,140 @@ If a format is listed in [[#plain-color-formats]] with {{GPUTextureUsage/RENDER_
+    ::
+        Indicates whether the object has been destroyed (can no longer be used).
+        Initially set to `false`.
+
+        <!-- Move this to an internal object if GPUExternalTexture becomes Serializable. -->
+</dl>
+
+### Importing External Textures ### {#external-texture-creation}
+
+An external texture is created from an external video object
+using {{GPUDevice/importExternalTexture()}}.
+
+Issue: Update this description with canvas.
+
+External textures are destroyed automatically, as a microtask,
+instead of manually or upon garbage collection like other resources.
+
+<script type=idl>
+dictionary GPUExternalTextureDescriptor : GPUObjectDescriptorBase {
+    required HTMLVideoElement source;
+    GPUPredefinedColorSpace colorSpace = "srgb";
+};
+</script>
+
+<dl dfn-type=method dfn-for=GPUDevice>
+    : <dfn>importExternalTexture(descriptor)</dfn>
+    ::
+        Creates a {{GPUExternalTexture}} wrapping the provided image source.
+
+        <div algorithm=GPUDevice.importExternalTexture>
+            **Called on:** {{GPUDevice}} this.
+
+            **Arguments:**
+            <pre class=argumentdef for="GPUDevice/importExternalTexture(descriptor)">
+                |descriptor|: Provides the external image source object (and any creation options).
+            </pre>
+
+            **Returns:** {{GPUExternalTexture}}
+
+            1. Let |source| be |descriptor|.{{GPUExternalTextureDescriptor/source}}.
+
+            1. If |source| <l spec=html>[=is not origin-clean=]</l>,
+                throw a {{SecurityError}} and stop.
+
+            1. Let |data| be the result of converting the current image contents of |source| into
+                the color space |descriptor|.{{GPUExternalTextureDescriptor/colorSpace}}.
+
+                Note: This is described like a copy, but may be implemented as a reference to
+                read-only underlying data plus appropriate metadata to perform conversion later.
+
+                Issue: It is currently undetermined whether the default colorSpace, "srgb", is
+                extended-srgb or clamped-srgb.
+                This will be determined upstream as the semantics around super-srgb image sources
+                get defined. Unfortunately we can't sidestep it for now because video sources
+                can go already go outside the srgb range.
+                The upstream determination will change whether using the default colorSpace
+                option can result in sampling values greater than 1.0 or not.
+                If upstream decides to make "srgb" mean clamped-srgb, we also have the option
+                of changing our default to `"extended-srgb"`.
+
+            1. Let |result| be a new {{GPUExternalTexture}} object wrapping |data|.
+
+            1. [=Queue a microtask=] to set |result|.{{GPUExternalTexture/[[destroyed]]}} to `true`,
+                releasing the underlying resource.
+
+                Issue: Is this too restrictive?
```

Comment:
so the intended use case is creating a new bind group each frame, given that there is a new `GPUExternalObject`?

Reply:
Yes, I don't think we can change this. The underlying textures are expected to come from external sources (like a video decoder) so we can't force them to always be the same texture object for the same source.